cosmwasm/contracts/crosschain-swaps/src/checks.rs

```diff
@@ -1,11 +1,32 @@
 use cosmwasm_std::{Addr, Deps};
+use regex::Regex;
 
 use crate::{state::CHANNEL_MAP, ContractError};
 
-// Validate that the receiver address is a valid address for the destination chain.
-// This will prevent IBC transfers from failing after forwarding
-pub fn validate_receiver(deps: Deps, receiver: Addr) -> Result<(String, Addr), ContractError> {
-    let Ok((prefix, _, _)) = bech32::decode(receiver.as_str()) else {
+/// If the specified receiver is an explicity channel+addr, extract the parts
+/// and use the strings as provided
+fn validate_explicit_receiver(receiver: &str) -> Result<(String, Addr), ContractError> {
+    let re = Regex::new(r"^ibc:(channel-\d+)/(.+)$").unwrap();
```

Comment:
nit/q: can this be global to avoid initializing it every time this is run?

Reply:
Yeah agreed, I think its important to have the regex compiled only once.

Reply:
Oh this might not be so simple: https://stackoverflow.com/questions/41503880/why-isnt-regex-a-wrapper-for-regexnew-to-offer-the-same-regex-matching-s

Reply:
Not seeing the regex! macro in latest docs, so maybe it just works as a global?

Reply:
It seems like the regex macro (1) didn't do what we want and (2) is gone: https://github.com/rust-lang/regex/blob/master/HACKING.md#the-regex-macro 

Using lazy_static! shouldn't help here, as we only call this once and IIUC the contract code gets loaded on each execution. 

I can replace the regex with manual checking. This is not hard and it's what I had before but I changed it to a regex to make the code shorter. it's basically: remove the `ibc:` prefix and check that it's there, split the string by `/` and check that the first part matches channel-x, use the rest as the denom (assuming addreses can't contain `/`)

Reply:
Found [this](https://news.ycombinator.com/item?id=27301320) which seems promising too, but it's not documented/standard, so we'd probably need to review the code.

Reply:
Cosmwasm tooling question -- if this doesn't exist, lets ask Boss to see if we can add it into beaker.

Can we use printlns to measure how much gas this call takes? This is really just a gas optimization question

Reply:
Lets make this a followup issue as gas optimization.

Reply:
Added https://github.com/osmosis-labs/osmosis/issues/4197 for this.